src/frontend/src/scheduler/plan_fragmenter.rs

```diff
@@ -751,6 +761,11 @@ impl BatchPlanFragmenter {
                     lookup_join_parallelism
                 } else if source_info.is_some() {
                     0
+                } else if let Some(num) = self.batch_parallelism {
+                    min(
+                        num.get() as usize,
+                        self.worker_node_manager.worker_node_count(),
```

Comment:
The goal here is to allow user to increase parallelism, and it should be smaller `worker_node_count * cores`